webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/EndpointsDefinition.scala

```diff
@@ -194,13 +194,29 @@ object NcfApi extends ApiModuleProvider[NcfApi] {
     val (action, path)  = GET / "techniques" / "{techniqueId}" / "{techniqueVersion}" / "resources"
   }
   final case object GetNewResources extends NcfApi with TwoParam with StartsAtVersion15 with SortIndex { val z = implicitly[Line].value
-    val description = "Get ressources of a new technique"
+    val description = "Get resources of a new technique"
     val (action, path)  = GET / "techniques" / "{techniqueId}" / "new" / "{techniqueVersion}" / "resources"
   }
   final case object ParameterCheck extends NcfApi with ZeroParam with StartsAtVersion15 with SortIndex { val z = implicitly[Line].value
     val description = "Get currently deployed resources of a technique"
     val (action, path)  = POST / "techniques" / "parameter" / "check"
   }
+  final case object GetTechniques extends NcfApi with ZeroParam with StartsAtVersion15 with SortIndex { val z = implicitly[Line].value
+    val description = "Get all Techniques metadata"
+    val (action, path)  = GET / "ncf" / "techniques"
+  }
+  final case object GetMethods extends NcfApi with ZeroParam with StartsAtVersion15 with SortIndex { val z = implicitly[Line].value
+    val description = "Get all methods metadata"
+    val (action, path)  = GET / "ncf" / "methods"
+  }
+  final case object UpdateMethods extends NcfApi with ZeroParam with StartsAtVersion15 with SortIndex { val z = implicitly[Line].value
+    val description = "Update methods metadata"
+    val (action, path)  = POST / "ncf" / "methods" / "update"
```

Comment:
Why no just `POST` on `ncf/methods` ? I think the update is not necessary and it's a break compared with what is used everywhere else in rudder

Reply:
up?

Reply:
I do think we need to have different url for this methods, it's not a POST where you give data directly, but here we start a reload of methods/techniques from the file system.

So i guess a better URL maybe POST /ncf/reload/(techniques|methods) would be better

Reply:
I also don't like that url prefix here ncf/

other api from this api (write a technique/ parameter check don't have this prefix)

Maybe we could have ncf prefix everywhere for this api ...